packages/toolkit/src/view/catalog/components/tabs/AskQuestionTab.tsx

```diff
@@ -0,0 +1,185 @@
+"use client";
+
+import * as React from "react";
+import { Nullable } from "instill-sdk";
+
+import { Separator } from "@instill-ai/design-system";
+
+import { CodeBlock, ModelSectionHeader } from "../../../../components";
+import { defaultCodeSnippetStyles } from "../../../../constant";
+import { Catalog } from "../../../../lib/react-query-service/catalog/types";
+import { env } from "../../../../server";
+
+type AskQuestionTabProps = {
+  catalog: Catalog;
+  isProcessed: boolean;
+  onGoToUpload: () => void;
+  namespaceId: Nullable<string>;
+};
+
+export const AskQuestionTab = ({
+  catalog,
+  isProcessed,
+  onGoToUpload,
+  namespaceId,
+}: AskQuestionTabProps) => {
```

Comment:
@thewbuk If the type is not exported or used in any other way, we can put it directly in the component 
```suggestion
export const AskQuestionTab = ({
  catalog,
  isProcessed,
  onGoToUpload,
  namespaceId,
}:{
 catalog: Catalog;
  isProcessed: boolean;
  onGoToUpload: () => void;
  namespaceId: Nullable<string>;
}) => {
```